### ISSUE TITLE
Add tryLast for Foldable

### ISSUE THREAD

**Issue by cmeeren** [fsprojects/FSharpPlus#327]
I see that `tryHead` is implemented, but `tryLast` is not. It would nicely complement the existing `tryHead`.

**Comment by wallymathieu** [comment]
My personal opinion is that it would make sense to add tryLast since there is `List.tryLast`.

**Comment by gusty** [comment]
How would it work with infinite sequences? I don't think we can make it safe in that case.

**Comment by wallymathieu** [comment]
That's true. That's one thing that speaks against adding it.

**Comment by cmeeren** [comment]
You have `Seq.tryLast` in F# code, so it doesn't seem to have stopped them from adding it.

**Comment by wallymathieu** [comment]
`Seq.tryLast` in f# core is probably to be compatible with common LINQ usage in .net.

**Comment by cmeeren** [comment]
I don't see how that changes anything; please enlighten me if I am missing something.

**Comment by wallymathieu** [comment]
I imagine that many usages of `seq` are more intended to wrap a finite collection rather than a potentially unbounded collection. In c# you sometimes have `seq` hiding a generator:

```c#
public IEnumerable<T> GetUnboundedResults()
{
        while(true)
        {
            var result = GenerateNextResult();
            yield return result;
        }
}
```

I've seen this pattern once or twice, but mostly people use `IEnumerable` in order to avoid leaking a specific collection implementation that they don't intend for you to modify (without creating a new collection based on it).

**Comment by gusty** [comment]
So, `tryLast` will be aimed to "fail" on empty structures, right?

**Comment by cmeeren** [comment]
Shouldn't `tryLast` return `None` for empty structures? That's what `Seq.tryLast`, `List.tryLast` etc. does.

**Comment by gusty** [comment]
Yes, that's what I meant with "fail", I was referring to the pure-failure, which in this case is `None`.

**Comment by gusty** [comment]
Implemented in #375
